core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -1041,7 +1042,11 @@ private[deploy] class Master(
       ApplicationInfo = {
     val now = System.currentTimeMillis()
     val date = new Date(now)
-    val appId = newApplicationId(date)
+    val appId = if (useAppNameAsAppId) {
+      desc.name.toLowerCase().replaceAll("\\s+", "")
+    } else {
+      newApplicationId(date)
+    }
```

Comment:
Hmm, how Spark responses to duplicate appId? If two submitted applications have same app names.

Reply:
Should we add `nextAppNumber` as postfix to the appId as usual?

Reply:
This experimental design is for the advanced users who maintains unique appName.